src/python/pants/backend/python/tasks2/pytest_run.py

```diff
@@ -77,15 +78,20 @@ def failed_targets(self):
     return self._failed_targets
 
 
-class PytestRun(TestRunnerTaskMixin, Task):
+class PytestRun(TestRunnerTaskMixin, PythonExecutionTaskBase):
 
   @classmethod
   def register_options(cls, register):
     super(PytestRun, cls).register_options(register)
-    register('--fast', type=bool, default=True,
-             help='Run all tests in a single pytest invocation. If turned off, each test target '
-                  'will run in its own pytest invocation, which will be slower, but isolates '
-                  'tests from process-wide state created by tests in other targets.')
+    register('--run-per-target', type=bool, default=False,
+             help='Invoke pytest for each target separately. This is slower, but isolates tests '
+                  'from process-wide state created by tests in other targets. If turned off, '
+                  'pytest will be invoked once for each subset in the partition. Each subset '
+                  'may contain multiple test targets. See --single-source-root for further '
+                  'control over test isolation.')
+    register('--single-source-root', type=bool, default=True,
```

Comment:
Not sure this is the best name for this, as "source root" is pre-existing terminology in Pants, and it's not what this is about.  Maybe something like "--global-resolve" ?